packages/plugins/src/foundry/utils/index.ts

```diff
@@ -1490,3 +1480,127 @@ const isStorageAccess = (
     typeof obj.reverted === 'boolean'
   )
 }
+
+export const validateProposalNetworks = async (
+  networks: Array<string>,
+  rpcEndpoints: FoundryToml['rpcEndpoints'],
+  isLiveNetwork: SphinxContext['isLiveNetwork']
+): Promise<{ rpcUrls: Array<string>; isTestnet: boolean }> => {
+  if (networks.length === 0) {
+    throw new Error(`Expected at least one network, but none were supplied.`)
+  }
+
+  const networkPromises = networks.map(async (network) => {
+    const rpcUrl = rpcEndpoints[network]
+    if (!rpcUrl) {
+      return { type: 'missingEndpoint', network }
+    }
+
+    const provider = new SphinxJsonRpcProvider(rpcUrl)
+    let networkInfo: ethers.Network
+    try {
+      networkInfo = await provider.getNetwork()
+    } catch {
+      return { type: 'failedRequest', network }
+    }
+
+    const sphinxNetwork = SPHINX_NETWORKS.find(
+      (supportedNetwork) => supportedNetwork.chainId === networkInfo.chainId
+    )
+
+    if (!sphinxNetwork) {
+      return {
+        type: 'unsupported',
+        network,
+        chainId: networkInfo.chainId.toString(),
+      }
+    }
+
+    if (!(await isLiveNetwork(provider))) {
+      return { type: 'localNetwork', network }
+    }
```

Comment:
Fyi, this check may break website tests if you don't pass in a mocked `isLiveNetwork`